zebra/redistribute.c

```diff
@@ -212,18 +221,9 @@ void redistribute_update(const struct prefix *p, const struct prefix *src_p,
 			}
 			zsend_redistribute_route(ZEBRA_REDISTRIBUTE_ROUTE_ADD,
 						 client, p, src_p, re);
-		} else if (prev_re
-			   && ((re->instance
-				&& redist_check_instance(
-					   &client->mi_redist[afi]
-							     [prev_re->type],
-					   re->instance))
-			       || vrf_bitmap_check(
-					  client->redist[afi][prev_re->type],
-					  re->vrf_id))) {
+		} else if (zebra_redistribute_check(prev_re, client, p, afi))
 			zsend_redistribute_route(ZEBRA_REDISTRIBUTE_ROUTE_DEL,
 						 client, p, src_p, prev_re);
-		}
```

Comment:
Here I am bit doubtfull on the checks.  For prev_re we are only checking for multi-instance and for given re-type and not a full check of default route.

Reply:
Indeed. I see no problem in checking it twice.

Reply:
It seems fine to check twice here "just in case;" if the check is removed, I would add a comment on why the check isn't needed so some future reader doesn't "miss" the check and think there is something wrong in some way.

Reply:
sure will add comments.